plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -205,123 +215,121 @@ func (plugin *T128Lte) getLteInterfaces() <-chan map[string]interface{} {
+		return LTEDeviceInfo{}, fmt.Errorf("Unable to get load data from %s", info.Name())
+	}
+	lteInfoContent, _ := ioutil.ReadFile(infoFile)
+	err := json.Unmarshal(lteInfoContent, &lteInfo)
 	if err != nil {
-		return nil, fmt.Errorf("LTE %s doesn't exists : %w", filePath, err)
+		return LTEDeviceInfo{}, fmt.Errorf("Cannot unmarshal LTE State Info %s : %w", lteInfo, err)
 	}
-	return checkFile, nil
+	return lteInfo, nil
 }
 
-func (plugin *T128Lte) getDeviceInterface(name string) (map[string]interface{}, error) {
-	content, err := os.ReadFile(T128_CONFIG_FILE)
-	if err != nil {
-		return nil, fmt.Errorf("Cannot read the %s : %w", content, err)
-	}
-	err = json.Unmarshal(content, &deviceInterfaceDetails)
+func getFileInfo(filePath string) (info fs.FileInfo, exists bool) {
+	info, err := os.Stat(filePath)
 	if err != nil {
-		return nil, err
-	}
-	for _, node := range deviceInterfaceDetails.Config.Authority.Router {
-		for _, deviceInterface := range node.Node {
-			for _, deviceType := range deviceInterface.DeviceInterface {
-				if deviceType.Type == name {
-					deviceDetails := map[string]interface{}{
-						"Name":            deviceType.Name,
-						"Type":            deviceType.Type,
-						"TargetInterface": deviceType.TargetInterface,
-					}
-					return deviceDetails, nil
-				}
-			}
-		}
+		return nil, false
 	}
-	return nil, fmt.Errorf("Cannot find the type")
+	return info, true
 }
 
-func (plugin *T128Lte) getServiceStatus(lteMetric map[string]interface{}) string {
-	if lteMetric["LTE"].(map[string]interface{})["Registration Status"].(string) == "registered" &&
-		lteMetric["LTE"].(map[string]interface{})["Connection Status"].(string) == "connected" {
+func (plugin *T128Lte) getServiceStatus(lteMetric LTEDeviceState) string {
+	if lteMetric.LTE.RegistrationStatus == "registered" && lteMetric.LTE.ConnectionStatus == "connected" {
 		return "normal"
 	}
-	if lteMetric["LTE"].(map[string]interface{})["failure"].(map[string]interface{})["reason"].(string) != "" {
-		registrationStatus := lteMetric["LTE"].(map[string]interface{})["failure"].(map[string]interface{})["reason"].(string)
-		return SERVICE_STATUS_MAP[registrationStatus]
+	if lteMetric.LTE.Failure.Reason != "" {
+		registerationStatus := lteMetric.LTE.Failure.Reason
+		return SERVICE_STATUS_MAP[registerationStatus]
 	}
-	return lteMetric["LTE"].(map[string]interface{})["Registration Status"].(string)
+	return lteMetric.LTE.RegistrationStatus
 }
 
-func (plugin *T128Lte) hasRequiredData(lteMetric map[string]interface{}) bool {
-	requiredData := []string{"Signal Strength", "RSSI Signal Strength"}
-	lteData, ok := lteMetric["LTE"].(map[string]interface{})
-	if !ok {
-		return false
-	}
+func (plugin *T128Lte) hasRequiredData(lteMetric LTEDeviceState) bool {
+	requiredData := []string{"SignalStrength", "RSSISignalStrength"}
+	lteValue := reflect.ValueOf(lteMetric)
 	for _, value := range requiredData {
-		if _, ok := lteData[value]; !ok {
+		field := lteValue.FieldByName(value)
```

Comment:
This is clever, but I don't think it's worth it for two fields. Just go ahead and check the fields.